### PR TITLE
Fix ModelFarm race conditions and unit tests

### LINKED ISSUE #1150: TestModelFarm: Racy unit test creates spurious failures in CI
```
io.deephaven.base.verify.RequirementFailure: Requirement failed: required terminated == true, instead terminated == false, testShutdownTimeoutSecs == 1.0.
	at io.deephaven.base.verify.Require.fail(Require.java:108)
	at io.deephaven.base.verify.Require.requirement(Require.java:247)
	at io.deephaven.base.verify.Require.requirement(Require.java:254)
	at io.deephaven.modelfarm.TestModelFarm.testModelFarmNoWorkShutdown(TestModelFarm.java:44)
```
The test in question sleeps for 100 milliseconds and assumes that's enough time for a condition to be met. This isn't a good strategy for reliable, consistent unit testing.

### LINKED ISSUE #1641: Race condition in ModelFarm shutdown
The ModelFarm shutdown methods have a race condition bug.

### PR DESCRIPTION
**Pull request by chipkent**
ModelFarm shutdown methods have a race condition that triggered occasional unit test failures.  Bugs were also found in the unit tests.

Resolves #1641 
Resolves #1150

### CODE REVIEW

File: ModelFarm/src/main/java/io/deephaven/modelfarm/ModelFarmBase.java [hunk 1]
```diff
@@ -340,8 +340,8 @@ public final boolean awaitTermination(final long timeout, final TimeUnit unit) {
                 timeout == Long.MAX_VALUE ? Long.MAX_VALUE : System.currentTimeMillis() + unit.toMillis(timeout);
         boolean allThreadsTerminated = false;
 
-        while (!allThreadsTerminated && System.currentTimeMillis() < timeoutMillis) {
-            synchronized (ModelFarmBase.this) {
+        synchronized (ModelFarmBase.this) {
+            while (!allThreadsTerminated && System.currentTimeMillis() < timeoutMillis) {
```
Comment (rcaudy) [inline ModelFarm/src/main/java/io/deephaven/modelfarm/ModelFarmBase.java:344]: You should record the current time (or the remaining time) from this line and use that in the `wait` below on L348. Currently you run the risk of an `IllegalArgumentException` if you ask to wait for a negative time, or blocking indefinitely if you ask to wait for zero milliseconds.

File: ModelFarm/src/test/java/io/deephaven/modelfarm/TestModelFarm.java [hunk 2]
```diff
@@ -22,7 +22,7 @@
 public class TestModelFarm extends TestCase {
 
     private final long testShutdownTimeoutSecs =
-            Configuration.getInstance().getIntegerWithDefault("TestModelFarm.testShutdownTimeoutSecs", 1);
+            Configuration.getInstance().getIntegerWithDefault("TestModelFarm.testShutdownTimeoutSecs", 5);
```
Comment (rcaudy) [inline ModelFarm/src/test/java/io/deephaven/modelfarm/TestModelFarm.java:25]: Were you planning to commit a bigger default? Did CI in fact time out, or was something else wrong?

File: ModelFarm/src/main/java/io/deephaven/modelfarm/ModelFarmBase.java (hunk 1 above)
Comment (chipkent) [reply ModelFarm/src/main/java/io/deephaven/modelfarm/ModelFarmBase.java]: Good catch.

File: ModelFarm/src/test/java/io/deephaven/modelfarm/TestModelFarm.java (hunk 2 above)
Comment (chipkent) [reply ModelFarm/src/test/java/io/deephaven/modelfarm/TestModelFarm.java]: I wasn't planning to increase the constant.  The bug was a branch that wasn't getting handled.  I just ran with the old constant and had zero failures for the first two (faster) test cases in >50k tries.

File: ModelFarm/src/main/java/io/deephaven/modelfarm/ModelFarmBase.java [hunk 3]
```diff
@@ -360,6 +362,8 @@ public final boolean awaitTermination(final long timeout, final TimeUnit unit) {
                 } else {
                     allThreadsTerminated = true;
                 }
+
+                currentTime = System.currentTimeMillis();
```
Comment (rcaudy) [inline ModelFarm/src/main/java/io/deephaven/modelfarm/ModelFarmBase.java:366]: Approving, but if you wanted this code to be as tight as you can make it I would either move to `System.currentTimeMillis()` to the where (that is, `while (!allThreadsTerminated && (currentTime = System.currenttimeMillis()) < timeoutMillis) {`) or add a `break` after the lines where `allThreadsTerminate = true;` is set.

**Review by rcaudy — APPROVED** [review]

**Pull request merged**